core/rawdb/accessors_chain.go

```diff
@@ -555,25 +555,18 @@ func ReadCXReceipts(db DatabaseReader, shardID uint32, number uint64, hash commo
 }
 
 // WriteCXReceipts stores all the transaction receipts given destination shardID, blockNumber and blockHash
-func WriteCXReceipts(db DatabaseWriter, shardID uint32, number uint64, hash common.Hash, receipts types.CXReceipts, temp bool) error {
+func WriteCXReceipts(db DatabaseWriter, shardID uint32, number uint64, hash common.Hash, receipts types.CXReceipts) error {
 	bytes, err := rlp.EncodeToBytes(receipts)
 	if err != nil {
 		utils.Logger().Error().Msg("[WriteCXReceipts] Failed to encode cross shard tx receipts")
 	}
 	// Store the receipt slice
-	if err := db.Put(cxReceiptKey(shardID, number, hash, temp), bytes); err != nil {
+	if err := db.Put(cxReceiptKey(shardID, number, hash), bytes); err != nil {
 		utils.Logger().Error().Msg("[WriteCXReceipts] Failed to store cxreceipts")
 	}
 	return err
 }
 
-// DeleteCXReceipts removes all receipt data associated with a block hash.
-func DeleteCXReceipts(db DatabaseDeleter, shardID uint32, number uint64, hash common.Hash, temp bool) {
```

Comment:
DeleteCXReceipts is never used. Then do we ever remove receipts that's not needed anymore?

Reply:
If we support the CXReceipt at any given time, then we should not delete the CXReceipt. Otherwise, we can clean the old CXReceipt when we know it's been delivered. We don't have the support to check and archive old CXReceipt yet.